packages/suite/src/components/suite/troubleshooting/TroubleshootingTips.tsx

```diff
@@ -1,160 +1,145 @@
-    font-size: ${variables.FONT_SIZE.TINY};
-    font-weight: ${variables.FONT_WEIGHT.MEDIUM};
-`;
-
-// eslint-disable-next-line local-rules/no-override-ds-component
-const StyledButton = styled(Button)`
-    margin: 0 20px 20px;
-`;
-
-interface Item {
+import { ReactNode, useState } from 'react';
+
+import {
+    Card,
+    Collapsible,
+    Column,
+    ElevationContext,
+    ElevationDown,
+    ElevationUp,
+    IconName,
+    Row,
+    SelectBar,
+    Text,
+} from '@trezor/components';
+import { spacings } from '@trezor/theme';
+
+import { TroubleshootingTipsFooter } from './TroubleshootingTipsFooter';
+import { TroubleshootingTipsListCard } from './TroubleshootingTipsListCard';
+import { TroubleshootingTipsToggle } from './TroubleshootingTipsToggle';
+
+export type TroubleshootingTipsItem = {
     key: string;
     heading?: ReactNode;
     description?: ReactNode;
     hide?: boolean;
     noBullet?: boolean;
     action?: ReactNode;
-}
+    icon?: IconName;
+};
 
-interface TroubleshootingTipsProps {
+type SectionDefinition = { label: ReactNode; items: TroubleshootingTipsItem[] };
+
+type TroubleshootingTipsWithSectionsProps<K extends string, T extends K> = {
     label: ReactNode;
     cta?: ReactNode;
-    items: Item[];
-    offerWebUsb?: boolean;
     opened?: boolean;
     'data-testid'?: string;
-}
+    items: Record<K, SectionDefinition>;
+    sectionLabel?: ReactNode;
+    defaultSection: T;
+    toggleText?: ReactNode;
+};
 
-export const TroubleshootingTips = ({
+export const TroubleshootingTipsWithSections = <K extends string, T extends K>({
     label,
     items,
     cta,
-    offerWebUsb,
     opened,
+    sectionLabel,
+    defaultSection,
+    toggleText,
     'data-testid': dataTest,
-}: TroubleshootingTipsProps) => {
-    const { elevation } = useElevation();
-
-    const memoizedItems = useMemo(
-        () =>
-            items
-                .filter(item => !item.hide)
-                .map(item => (
-                    <Item key={item.key}>
-                        {!item.noBullet && <Bullet>&bull;</Bullet>}
-
-                        <ItemContent>
-                            <ItemLabel>{item.heading}</ItemLabel>
-                            <ItemDescription>{item.description}</ItemDescription>
-                        </ItemContent>
-
-                        {item.action && <ItemAction>{item.action}</ItemAction>}
-                    </Item>
-                )),
-        [items],
+}: TroubleshootingTipsWithSectionsProps<K, T>) => {
+    const [isOpened, setIsOpened] = useState(opened === true);
+    const [selectedSection, setSelectedSection] = useState<K>(defaultSection ?? null);
+
+    const headingRow = (
+        <Row
+            justifyContent="space-between"
+            alignItems="center"
+            margin={{ horizontal: spacings.sm }}
+        >
+            <Text typographyStyle="body">
+                {Object.keys(items).length > 1 ? sectionLabel : label}
```

Comment:
What about extracting `Object.keys(items).length > 1` to some const? You use it twice and it could improve readability